starknet_py/tests/e2e/declare/declare_test.py

```diff
@@ -13,3 +13,24 @@ async def test_declare_tx(account, map_compiled_contract):
     await account.client.wait_for_tx(
         tx_hash=result.transaction_hash, wait_for_accept=True
     )
+
+
+@pytest.mark.asyncio
+async def test_declare_v2_tx(account, sierra_minimal_compiled_contract_and_class_hash):
+    (
+        compiled_contract,
+        compiled_class_hash,
+    ) = sierra_minimal_compiled_contract_and_class_hash
+
+    declare_tx = await account.sign_declare_transaction(
+        compiled_contract=compiled_contract,
+        compiled_class_hash=compiled_class_hash,
+        max_fee=MAX_FEE,
+    )
+    assert declare_tx.version == 2
+
+    result = await account.client.declare(declare_tx)
+
+    await account.client.wait_for_tx(
+        tx_hash=result.transaction_hash, wait_for_accept=True
+    )
```

Comment:
What about an assert here? Shouldn't we check if the transaction passed?

Reply:
`wait_for_tx` with `wait_for_accept` will raise if the transaction doesn't get the `ACCEPTED_ON_L2` status. Assert doesn't seem necessary here.